app/hooks/useMountEffect.js

```diff
@@ -6,8 +6,12 @@ import { useEffect } from "react";
  * only after succesful mount (empty array as second argument
  * in useEffect call)
  *
+ * Note: eslint react-hooks/exhaustive-deps is disabled here due to this being
+ * specifically to run a function on the very first mount, independently of fun
+ * changing, thus `fun` is not supposed to actually be a dependency.
+ *
  * @param {*} fun - effect to run on mount
  */
-const useMountEffect = (fun) => useEffect(fun, []);
+const useMountEffect = (fun) => useEffect(fun, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
hrmm i'm totally fine with disabling eslint here if it's complaining but i'm just wondering as we always had `react-hooks/exhaustive-deps` on but it never complained about this one and tbh it shouldn't.
the `fun` here is a function param which means it defined inside the scope of `useMountEffect` hence i would expect eslint to not complain that it's not in the deps array.

Reply:
It started complaining after updating one of the dependencies, so a newer linter version is probably being more aggressive in detecting these requirements.

But I'm confused by this:

> the fun here is a function param which means it defined inside the scope of useMountEffect

I might be parsing this phrase wrongly, but I'm not sure this statement is correct. `const useMountEffect = /*...*/` is _not_ a function definition, but an assignment of a closure.

In any case, it seems to me the linter is in fact correct in complaining here. Consider:

```js
function MyCompo() {
  const [val, setVal] = useState('');
  let f = () => console.log("bleh");
  if val != '' {
    f = () => console.log("blah");
  }

  useMountEffect(f);
  
  // The above is equivalent to
  // useEffect(f, []);

  //...
}
```

This is wrong in all sorts of ways, but the key bit is that the contents of `f` var can change arbitrarily between renders (for example, due to the value of the `val` state), and thus it must be part of the dependency list for the `useEffect` call.

Reply:
Hrmm I see what you mean, but I think the deps array is about deps of `f` not `f` itself, which in the example above is the actualt _effect_ and not one of it's deps. 

If f was `() => console.log(val)` I would expect val to be in the deps array but that isn't the case in the example above.

Reply:
I agree with your assessment (which is why the warning is also somewhat confusing to me).

I don't know enough about the underlying React and linter behavior to know for sure the origin of the warning, the above was my guess at what was happening based on `useMountEffect` using a parametrized function (vs an inline one).